flanker/addresslib/address.py

```diff
@@ -390,27 +547,61 @@ def supports_routing(self):
 
     def full_spec(self):
```

Comment:
It will be as unexpected when method succeeds but the returned result cannot be used in the same context. E.g. when an email is sent to a SMTP server that does not support UTF8 or the result is parsed with a regex that only expects only ASCII characters. So it is beneficial to fail fast. Don't you think?